packages/eas-json/tsconfig.build.json

```diff
@@ -1,4 +1,8 @@
 {
   "extends": "./tsconfig",
+  "compilerOptions": {
+    "noUnusedLocals": true,
+    "noUnusedParameters": true
+  },
```

Comment:
Most people use `yarn watch` and `yarn build` commands for local development. These commands use tsconfig.build.json, so, unfortunately, we don't resolve the issue here.
<img width="766" alt="Screenshot 2023-07-17 at 16 22 04" src="https://github.com/expo/eas-cli/assets/55145344/8de5f67d-69d4-490f-b76d-929b20e4385a">

What I meant was to create 2 different sets of config `local` and `production/ci` with different `tsconfig` and `tsconfig.build`